src/Servers/Kestrel/Core/src/KestrelServer.cs

```diff
@@ -57,4 +60,26 @@ public void Dispose()
     {
         _innerKestrelServer.Dispose();
     }
+
+    private sealed class SimpleHttpsConfigurationHelper : IHttpsConfigurationHelper
```

Comment:
I did the exact same thing in my metrics PR. It's annoying that a thing designed for DI - Kestrel - has a way of creating it without DI.

@Tratcher @halter73 @davidfowl Do we know how popular this type is? Could we mark it as obsolete and make it internal in the future?